rpc/rest/security/rest_security.go

```diff
@@ -0,0 +1,304 @@
+	tokenDb map[string]string
+
+	// Token claims
+	expTime int
+}
+
+// NewAuthenticator prepares new instance of authenticator.
+func NewAuthenticator(router *mux.Router, expTime int, users []httpsecurity.User, cost int, log logging.Logger) AuthenticatorAPI {
+	a := &authenticator{
+		router: router,
+		log:    log,
+		formatter: render.New(render.Options{
+			IndentJSON: true,
+		}),
+		userDb:  make([]*httpsecurity.User, 0),
+		tokenDb: make(map[string]string),
+		expTime: expTime,
+	}
+
+	// Add admin-user, enabled by default, always has access to every URL
+	users = append(users, httpsecurity.User{
+		Name:        admin,
+		Password:    "ligato123",
+		Permissions: []string{admin},
+	})
+
+	go func() {
+		// Process users in go routine, since hashing may take some time
+		for _, user := range users {
+			// Password hash cost is currently set via config file. Values allowed by bcrypt are in range 4-31. The min
+			// value will be set if cost is not defined in config file. Keep in mind that high cost values (>10) can
+			// take a lot of time to process.
+			hash, err := bcrypt.GenerateFromPassword([]byte(user.Password), cost)
+			if err != nil {
+				a.log.Errorf("failed to hash password for user %s: %v", user.Name, err)
+				continue
+			}
+
+			a.userDb = append(a.userDb, &httpsecurity.User{
+				Name:        user.Name,
+				Password:    string(hash),
+				Permissions: user.Permissions,
+			})
+			a.log.Debugf("Registered user %s, permissions: %v", user.Name, user.Permissions)
+		}
+	}()
+
+	// Admin-group, available by default and always enabled for all URLs
+	a.AddPermissionGroup(&httpsecurity.PermissionGroup{
+		Name: admin,
+	})
+
+	a.registerSecurityHandlers()
+
+	return a
+}
+
+// AddPermissionGroup adds new permission group.
+func (a *authenticator) AddPermissionGroup(group ...*httpsecurity.PermissionGroup) {
+	for _, newPermissionGroup := range group {
+		for _, existingGroup := range a.groupDb {
+			if existingGroup.Name == newPermissionGroup.Name {
+				a.log.Warnf("permission group %s already exists, skipped")
+				continue
+			}
+		}
+		a.log.Debugf("added HTTP permission group %s", newPermissionGroup.Name)
+		a.groupDb = append(a.groupDb, newPermissionGroup)
+	}
+}
+
+// Validate the request
+func (a *authenticator) Validate(provider http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		authHeader := req.Header.Get(authHeaderStr)
+		if authHeader == "" {
+			a.formatter.Text(w, http.StatusUnauthorized, "401 Unauthorized: authorization header required")
+			return
+		}
+		bearerToken := strings.Split(authHeader, " ")
+		if len(bearerToken) != 2 {
+			a.formatter.Text(w, http.StatusUnauthorized, "401 Unauthorized: invalid authorization token")
+			return
+		}
+		token, err := jwt.Parse(bearerToken[1], func(token *jwt.Token) (interface{}, error) {
+			if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
```

Comment:
Please use [GetSigningMethod](https://godoc.org/github.com/dgrijalva/jwt-go#GetSigningMethod) to validate this.